source/app/views/urls/edit.html.erb

```diff
@@ -0,0 +1,15 @@
+<%= form_for @url do |f| %>
+  <p>
+    http://fa.ke/<%= @url.short %>
+  </p>
+  <p>
+    <%= f.label :full, 'URL:' %>
+    <%= f.text_field :full %>
+    <% unless @url.errors[:full].empty? %>
+      <%= content_tag(:div, "Url #{@url.errors[:full][0]}", class: 'error-msg') %>
```

Comment:
This error handling part is kinda unfortunate. Perhaps a view helper or partial could clean it up? Using [SimpleForm](https://github.com/plataformatec/simple_form) might be heavy-handed for this little app ...